subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -249,8 +249,9 @@ do_firmware_transfer_reply_cb(const struct coap_packet *response,
 		SYS_LOG_ERR("Unexpected response from server: %d.%d",
 			    COAP_RESPONSE_CODE_CLASS(resp_code),
 			    COAP_RESPONSE_CODE_DETAIL(resp_code));
-		lwm2m_firmware_set_update_result(RESULT_CONNECTION_LOST);
```

Comment:
We used to set result to RESULT_CONNECTION_LOST here.  Now it's -ENOENT which isn't handled directly below so it falls back to RESULT_UPDATE_FAILED.

Reply:
I'm going to use "-ENOMSG /* Unexpected message type */" to represent the case.
ENOENT seems a common error number and we might have other issues if we map it to connection lost.
Or maybe we just set the state and return here directly.